### ISSUE TITLE
[CI] Octave 4.2 build failing at package install time

### ISSUE THREAD

**Issue by ojwb** [swig/swig#909]
The CI build for Octave 4.2 has started to fail, apparently every time.  For example, tail of log from https://travis-ci.org/swig/swig/jobs/203229070:

```
Setting up octave4.2-common (4.2.0-0+kwwette+trusty17) ...
Setting up octave4.2 (4.2.0-0+kwwette+trusty17) ...
/usr/bin/octave-cli: error while loading shared libraries: libGraphicsMagick++.so.3: cannot open shared object file: No such file or directory
dpkg: error processing package octave4.2 (--configure):
 subprocess installed post-installation script returned error exit status 127
dpkg: dependency problems prevent configuration of liboctave4.2-dev:
 liboctave4.2-dev depends on octave4.2 (= 4.2.0-0+kwwette+trusty17); however:
  Package octave4.2 is not configured yet.
dpkg: error processing package liboctave4.2-dev (--configure):
 dependency problems - leaving unconfigured
Setting up pstoedit (3.62-1ubuntu1) ...
No apport report written because the error message indicates its a followup error from a previous failure.
Setting up libauthen-sasl-perl (2.1500-1) ...
Setting up libwww-perl (6.05-2) ...
Setting up liblwp-protocol-https-perl (6.04-2ubuntu0.1) ...
Setting up libxml-parser-perl (2.41-1build3) ...
Setting up libxml-sax-expat-perl (0.40-2) ...
update-perl-sax-parsers: Registering Perl SAX parser XML::SAX::Expat with priority 50...
update-perl-sax-parsers: Updating overall Perl SAX parser modules info file...
Replacing config file /etc/perl/XML/SAX/ParserDetails.ini with new version
Processing triggers for libc-bin (2.19-0ubuntu6.9) ...
Errors were encountered while processing:
 octave4.2
 liboctave4.2-dev
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

Assigning to kwwette as it seems the packages are from him/her judging by the version number (4.2.0-0+kwwette+trusty17).

This regression is annoying as it means the CI status is always "failed" and one has to manually check if it's just this build which has failed.  If it's not going to be fixed quickly I'm going to disable this build for the time being.

**Comment by ojwb** [comment]
Just remembered we have a list of builds for which failure is allowed, so I've just moved octave 4.2 there for now.

**Comment by kwwette** [comment]
I'm not sure the `allow_failures` section in `.travis.yml` is correct: from https://docs.travis-ci.com/user/customizing-the-build#Rows-that-are-Allowed-to-Fail it looks like jobs must be included _both_ in `matrix` _and_ in `allow_failures`, as is the case for the `env: SWIGLANG=python SWIG_FEATURES=-O` job. At the moment the `SWIGLANG=octave SWIGJOBS=-j2 VER=4.2 CPP11=1` job isn't being run at all. This is also the case for the `SWIGLANG=ocaml` job.

**Comment by ojwb** [comment]
Hmm, so you're forced to duplicate the config for such jobs - what a dumb design!

**Comment by ojwb** [comment]
@kwwette But the real issue here is that the job has started failing every time - we're using packages from your PPA so I was hoping you'd have some insight as to why that might be.

**Comment by kwwette** [comment]
@ojwb - I'll have a look at what's going on with the PPA. It was working before so I'm not sure exactly what could have broken...

**Comment by kwwette** [comment]
The reason seems to be that the Travis environment is pulling in packages from this PPA:
https://launchpad.net/~rwky/+archive/ubuntu/graphicsmagick/+packages
This PPA is supplying newer packages for GraphicsMagick than exist in Trusty, which my PPA build of Octave links against. So Octave ends up with a newer version of `libGraphicsMagick++.so.*` than it was linked against, hence the error.

I'm don't know why this is happening however. I though Travis was supposed to supply a clean build environment, so it's a little alarming that it's pulling packages from some PPA that wasn't asked for! (I assume SWIG doesn't ask for this PPA, as I can't see any reference to it in git.) I also don't want to rebuild my PPA Octave against some PPA that Travis supplies, and which I might not trust.

@wsfulton - any idea why this might be happening?

**Comment by kwwette** [comment]
I think I have a fix for this now, please look at merge request #949

**Comment by wsfulton** [comment]
Closed by #949
